app/Http/Controllers/Auth/TwoFactorController.php

```diff
@@ -99,10 +102,24 @@ public function confirm(Request $request)
 
     public function redirectToExpected(Request $request)
     {
+        $this->rememberDevice($request);
         // Assuming 2fa passes, redirect to the expected url and remove it from session.
-        // NOTE: the url.expected is set in app\Http\Middleware\Google2FA.php
+        // NOTE: the url.expected is set in app\Http\Middleware\Google2FA.php.
         $expectedUrl = session()->get('url.expected');
         session()->remove('url.expected');
+
         return redirect($expectedUrl);
     }
+
+    protected function rememberDevice(Request $request)
+    {
+        $remember = $request->input('remember_device');
+        if ($remember) {
+            Cookie::queue(
+                'remember_device',
+                'yes',
```

Comment:
Not entirely sure what to use as the content for the cookie. It's hashed/encrypted client side, and we pretty much just need to check for it's existence. We could do something similar to the "remember password" like storing a token in the user table and validating against that.

Reply:
I think we should do the same as the "remember password" token. Right now its not user specific - once I've checked "remember this device" as one user, I can log in as another user and still skip 2fa.

Reply:
I've been looking at how Laravel uses that remember password token/cookie (in `vendor\laravel\framework\src\Illuminate\Auth\SessionGuard.php`). It cycles the token regularly (on every logout) which we should consider as well. It would help stop someone from stealing the cookie and using it indefinitely. 

Some other thoughts:
- There's nothing stopping someone from copying the cookie onto another device.
- Using the cookie like this would mean only one user can be remembered per device (probably fine).